### PR TITLE
Unit test/discover

### PR DESCRIPTION
**Pull request by Treesa-James**

### CODE REVIEW

File: tests/IntegrationTest.php [hunk 1]
```diff
@@ -56,109 +57,160 @@ function testDiscover()
...
-        $this->assertEquals(JsonSchemaFormats::None, $schema->getDataType('company','name')->format);
+        $this->assertTrue($schema->hasProperty('companies','domain'));
+        $this->assertEquals('string',$schema->schema['items']['companies']['properties']['domain']['type']);
 
-        $this->assertTrue($schema->hasProperty('company','domain'));
-        $this->assertEquals(JsonSchemaTypes::String, $schema->getDataType('company','domain')->type);
-        $this->assertEquals(JsonSchemaFormats::None, $schema->getDataType('company','domain')->format);
+        //Check if  required properties exist in companies and also have expected datatypes
+        $this->assertTrue($schema->hasProperty('companies','city'));
+        $this->assertEquals('string',$schema->schema['items']['companies']['properties']['name']['type']);
```
Comment (arun-experion) [inline tests/IntegrationTest.php:113]: Correct this

**Review by arun-experion — COMMENTED** [review]
Review comments added

File: tests/IntegrationTest.php (hunk 1 above)
Comment (Treesa-James) [reply tests/IntegrationTest.php]: corrected

File: tests/IntegrationTest.php [hunk 2]
```diff
@@ -56,109 +57,140 @@ function testDiscover()
         // Assert that the encoded JSON schema is a valid JSON string
         $this->assertJson($jsonSchema);
         // Compare the JSON schema with the expected schema stored in a file
-        $this->assertTrue(file_get_contents(__DIR__ . "/testDiscover.json") === $jsonSchema, "Schema is different than what was expected.");
+        $this->assertTrue(file_get_contents(__DIR__ . "DiscoverResult.json") === $jsonSchema, "Schema is different than excepted.");
```
Comment (arun-experion) [inline tests/IntegrationTest.php:60]: copy this file to __DIR__ . "/schemas/DiscoverResult.json and change the path

**Review by arun-experion — COMMENTED** [review]
Review comment added

**Pull request merged**